internal/filter/filter.go

```diff
@@ -0,0 +1,41 @@
+package filter
+
+import (
+	"sync/atomic"
+
+	"github.com/Azure/azure-storage-fuse/v2/internal"
+)
+
+var GlbFilterArr [][]Filter //it will store the fliters, outer array splitted by ||, inner array splitted by &&
+
+func ApplyFilterOnBlobs(fileInfos []*internal.ObjAttr) []*internal.ObjAttr { //function called from azstorage.go streamDir func
```

Comment:
From outside it shall be an object based call.
- read config as a string from cli
- parse it and create a filter object
- filtering code shall be executed on this object
- if filters are not applied then object shall be nil
- if not nil then apply the filters and it will return back a new slice.
- this model will allow us to have multiple instances of filter across the code if required,